backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardlist/BpkCardList.kt

```diff
@@ -0,0 +1,50 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.cardlist
+
+import androidx.compose.foundation.lazy.LazyItemScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.cardlist.internal.BpkCardListImpl
+import net.skyscanner.backpack.compose.sectionheader.BpkSectionHeaderButton
+
+@Composable
+fun BpkCardList(
+    title: String,
+    description: String,
+    layout: BpkCardListLayout,
+    modifier: Modifier = Modifier,
+    headerButton: BpkSectionHeaderButton? = null,
+    initiallyShownCards: Int = 3,
```

Comment:
this is something not very clear in the api table, but this only really makes sense in the stack version where there is an expand option.
i think we still need to expose a state, like in carousel with the size of the items, current position etc otherwise the component won't know how many items there are

Reply:
any suggestion how should we proceed here?

Reply:
So I think the best approach that will be backwards compatible for when we add the stack variant as well is to actually have a `BpkRailCardList` composable (similar to how we have the single vs multi select chip group) and remove the `layout` param.
Thinking through the API for the stack version again I think this will need to look quite different to be able to support nesting scrolling etc (we won't be able to have a `LazyColumn` internally as you can't nest `LazyColumn`s with flexible height)

For the rail variant internally the card list should use a `HorizontalPager` with the `pageSize` set (it's worth looking at `HotelMapCardsCarousel` in the app for that) - that will then set the size of the items correctly and do the docking. Again - worth double checking with design if that's what we want or not, but I suspect we do :)
For the state: we can then reuse the `BpkCarouselState` and add it as a parameter in the API. However it is worth double checking with design again if we want to have the endless scrolling here as well. If not we can just expose the default pager state.
Give me a shout if you have more questions - happy to pair on this.

Reply:
```
So I think the best approach that will be backwards compatible for when we add the stack variant as well is to actually have a BpkRailCardList composable (similar to how we have the single vs multi select chip group) and remove the layout param.
```
Great point, I'll migrate the code for this style
![Screenshot 2024-05-29 at 15 43 22](https://github.com/Skyscanner/backpack-android/assets/12631700/874934d0-80d6-4621-89b3-c0e4f446d41e)

about the `HorizontalPager`, I'll check